### PR TITLE
Get application files to Rails 4.2 standard

### PR DESCRIPTION
**Pull request by h-lame**
I ran `rake rails:update` to update existing files and generate new ones. This means our app is now configured like a Rails 4.2 app should be, and this will make it easier to upgrade to Rails 5.

This commit also migrates the app to use `secrets.yml`, and importantly this means it uses the `SECRET_KEY_BASE` env var on deployed environments. The infrastructure work to support this has already been done so we can deploy this without further work. Prior to this we relied on the
`secret_token.rb` initializer being overwritten by our deployment process, but that functionality was stopped in Dec 2016, meaning the app has been insecure since then.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -1,73 +1,30 @@
...
-    # Enable the asset pipeline
-    config.assets.enabled = true
-
-    # Version of your assets, change this if you want to expire all your assets
-    config.assets.version = '1.0'
-    config.assets.prefix = '/assets'
 
-    config.assets.precompile += %W(html5.js)
+    # Do not swallow errors in after_commit/after_rollback callbacks.
+    config.active_record.raise_in_transactional_callbacks = true
```
Comment (gpeng) [inline config/application.rb]: I think this is causing the test failure because `active_record` isn't loaded.

File: config/locales/en.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# To use a different locale, set it with `I18n.locale`:
+#
+#     I18n.locale = :es
+#
+# This would use the information in config/locales/es.yml.
+#
+# To learn more, please read the Rails Internationalization guide
+# available at http://guides.rubyonrails.org/i18n.html.
+
+en:
```
Comment (gpeng) [inline config/locales/en.yml]: Do we need this? In fact, do we need this file?

Comment (h-lame) [inline config/locales/en.yml]: Do we need it; probably not.  Is it worth having so the diff for future rails upgrades is minimal and we don't need to worry about it again; I'd say so, yes.

File: config/application.rb (hunk 1 above)
Comment (h-lame) [inline config/application.rb]: d'oh!  Thanks @gpeng !

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  puts "== Installing dependencies =="
+  system "gem install bundler --conservative"
+  system "bundle check || bundle install"
+
+  # puts "\n== Copying sample files =="
+  # unless File.exist?("config/database.yml")
+  #   system "cp config/database.yml.sample config/database.yml"
+  # end
+
+  puts "\n== Preparing database =="
```
Comment (andrewgarner) [inline bin/setup]: We should comment out these steps as we're not using ActiveRecord.

File: config/environments/development.rb [hunk 4]
```diff
@@ -1,30 +1,41 @@
...
   config.action_mailer.raise_delivery_errors = false
 
-  # Print deprecation notices to the Rails logger
+  # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
-  # Only use best-standards-support built into browsers
-  config.action_dispatch.best_standards_support = :builtin
+  # Raise an error on page load if there are pending migrations.
+  config.active_record.migration_error = :page_load
```
Comment (andrewgarner) [inline config/environments/development.rb]: We'll need to comment this out.

File: config/environments/production.rb [hunk 5]
```diff
@@ -1,67 +1,80 @@
...
 
-  # Send deprecation notices to registered listeners
+  # Send deprecation notices to registered listeners.
   config.active_support.deprecation = :notify
 
+  # Use default logging formatter so that PID and timestamp are not suppressed.
+  config.log_formatter = ::Logger::Formatter.new
+
+  # Do not dump schema after migrations.
+  config.active_record.dump_schema_after_migration = false
```
Comment (andrewgarner) [inline config/environments/production.rb]: And this.

File: config/locales/en.yml (hunk 2 above)
Comment (gpeng) [inline config/locales/en.yml]: Yep... fair enough.

Comment (andrewgarner) [inline config/locales/en.yml]: We don't need it, but the comments at the top of the file are useful to anyone who might look to add future translations. It also makes the upgrade process to Rails 5 easier as out application will be closer to that off a scaffolded application and thus the diff should be simpler.

File: config/application.rb (hunk 1 above)
Comment (h-lame) [inline config/application.rb]: commented out all active_record config options and rebased.  I figure that leaving them in, but commented out, is best so the diff for future upgrades is easier to understand.

File: bin/setup (hunk 3 above)
Comment (h-lame) [inline bin/setup]: Oh, yes.  Will do.

Comment (h-lame) [inline bin/setup]: Done and rebased.

**Pull request merged**